src/neptune/new/types/atoms/file.py

```diff
@@ -42,81 +40,80 @@
     is_plotly_figure,
 )
 from neptune.new.types.atoms.atom import Atom
+from neptune.new.types.atoms.file_types import _FileComposite
+from neptune.new.types.atoms.file_types import _FileType as FileType
+from neptune.new.types.atoms.file_types import (
+    _InMemoryComposite,
+    _LocalFileComposite,
+    _StreamComposite,
+)
 
 if TYPE_CHECKING:
     from neptune.new.types.value_visitor import ValueVisitor
 
 Ret = TypeVar("Ret")
 
 
-class FileType(enum.Enum):
-    LOCAL_FILE = "LOCAL_FILE"
-    IN_MEMORY = "IN_MEMORY"
-
-
 class File(Atom):
-    def __init__(
-        self,
-        path: Optional[str] = None,
-        content: Optional[bytes] = None,
-        extension: Optional[str] = None,
-    ):
+    def __init__(self, path: Optional[str] = None, file_composite: Optional[_FileComposite] = None):
+        """We have to support `path` parameter since almost all of `File` usages by our users look like `File(path)`."""
         verify_type("path", path, (str, type(None)))
-        verify_type("content", content, (bytes, type(None)))
```

Comment:
`File` is user-exposed and it's pretty common pattern to init it with `File(....)`. I think that dropping `content` and `extension` is a breaking change. Something went wrong here :wink:

Reply:
I've discussed it with @Herudaio, he's ok with that.